CRM/Core/BAO/MessageTemplate.php

```diff
@@ -342,7 +342,11 @@ public static function sendTemplate($params) {
     // If you want to expose the model via hook, consider interjecting a new Hook::alterWorkflowMessage($model) between `importAll()` and `exportAll()`.
 
     $params = array_merge($modelDefaults, $viewDefaults, $envelopeDefaults, $params);
-
+    if (!isset($params['workflow_name'])) {
+      // This is the parameter we recommend hooks listen to as it
+      // the option group is obsolete.
+      $params['workflow_name'] = $params['valueName'] ?? NULL;
```

Comment:
On naming: Based on auditing the parameters that come through here (#21615; eg https://github.com/civicrm/civicrm-core/pull/21615/commits/650646ea3700962100ec11fae274b6cba6356159), it is most correct to describe the naming-convention as "*`lowerCamel` xor `Upper-Kebab`*". Specifically, the tally for each notation is:

* `lowerCamel`: 19x params
* `lower` (one word): 11x (*compatible with `lowerCamel` or `lower-kebab` or `lower_snake` or `lowersmash`*)
* `Upper-Kebab`: 8x (*these are passthru email headers*)
* `lower_snake`: 2x
* `UpperCamel`: 1x (PDFFilename)

Sothis  should be called `workflowName` or just `workflow`. (*I'd encourage `workflow` as a way to evoke the constants `WORKFLOW` from `CRM_*_WorkflowMessage_*`.)

Reply:
@totten I'm fine with workflow - my general expection is that when passing a database field we use the database field name - but I'm also ok with shortening to workflow per your analysis (updated)